### ISSUE TITLE
`RequestTransformFunction` `resourceType` argument allows `undefined`  but it isn't clear what this means, beause there's also `ResourceType.Unknown`

### ISSUE THREAD

**Issue by ghost** [maplibre/maplibre-gl-js#3195]
The typing is here:

https://github.com/maplibre/maplibre-gl-js/blob/be14c414d896ca0c3fd71d37c93bc65a157fdedb/src/util/request_manager.ts#L21

The ambiguity (`Unknown` being an allowed value) is here:

https://github.com/maplibre/maplibre-gl-js/blob/be14c414d896ca0c3fd71d37c93bc65a157fdedb/src/util/request_manager.ts#L6-L15

If `undefined` means the same as `ResourceType.Unknown` then I'd suggest to make this a mandatory argument (so transform-request handlers don't have to convert `undefined` to `ResourceType.Unknown` on their own).
If those two aren't the same, then it should be documented what it means.

This caused an issue in another project for me: https://github.com/rowanwins/maplibregl-mapbox-request-transformer/issues/7

**Comment by HarelM** [comment]
I don't think you can call this method without this parameter from maplibre's code itself:
https://github.com/maplibre/maplibre-gl-js/blob/be14c414d896ca0c3fd71d37c93bc65a157fdedb/src/util/request_manager.ts#L37C10-L37C10

You can specify a method that doesn't do anything with this parameter I believe.
I'm not sure I fully understand the problem here.
Can you create a jsbin/codepen/stackblitz to illustrate the issue?

**Comment by HarelM** [comment]
I don't have hard feelings either way, feel free to submit a PR.
